api/types/header/convert/v1/header.go

```diff
@@ -67,11 +65,19 @@ func FromMetadataProto(msg *headerv1.Metadata) header.Metadata {
 
 // ToMetadataProto converts an internal metadata object into a v1 metadata protobuf message.
 func ToMetadataProto(metadata header.Metadata) *headerv1.Metadata {
+	// We map the zero go time to the zero protobuf time (nil).
+	var expires *timestamppb.Timestamp
+	if metadata.Expires.IsZero() {
+		expires = nil
+	} else {
+		expires = timestamppb.New(metadata.Expires)
+	}
```

Comment:
`expires` is `nil` anyways
```suggestion
	if !metadata.Expires.IsZero() {
		expires = timestamppb.New(metadata.Expires)
	}
```